packages/@sanity/default-layout/src/navbar/search/contexts/search/index.tsx

```diff
@@ -76,25 +83,40 @@ export function SearchProvider({children, currentUser}: SearchProviderProps) {
    * There are cases were we may not run searches when terms change (e.g. when search terms are empty / invalid).
    */
   useEffect(() => {
-    const termsChanged = Object.keys(terms).some(
-      (key) => terms[key] !== previousTermsRef.current?.[key]
-    )
+    const orderingChanged = !isEqual(ordering, previousOrderingRef.current)
     const pageIndexChanged = pageIndex !== previousPageIndexRef.current
+    const termsChanged = !isEqual(terms, previousTermsRef.current)
+
+    if (orderingChanged || pageIndexChanged || termsChanged) {
+      // Use a custom label if provided, otherwise return field and direction, e.g. `_updatedAt desc`
+      const sortLabel =
+        ordering?.customMeasurementLabel || `${ordering.sort.field} ${ordering.sort.direction}`
 
-    if (termsChanged || pageIndexChanged) {
       handleSearch({
-        ...terms,
-        limit: SEARCH_LIMIT,
-        offset: pageIndex * SEARCH_LIMIT,
+        options: {
+          // Comments prepended to each query for future measurement
+          comments: [
```

Comment:
Consider extracting the comments constructuin and adding some unit tests. Since its invisible to the user, we might not pick up if we break it (for instant if isRecentSearchTerms is changed without thinking about this usage).

Reply:
Would love a bit more direction here on specific tests you'd expect!

Right now, `comments` in search options has type safety, and there's [a test for createSearchQuery](https://github.com/sanity-io/sanity/blob/f698c924f9c26783e751bccc4f78e4d44171a273/packages/%40sanity/base/src/search/weighted/createSearchQuery.test.ts#L189-L203) that (loosely) covers prepended comments when creating queries